packages/dev/core/src/Inputs/scene.inputManager.ts

```diff
@@ -263,6 +264,41 @@ export class InputManager {
         }
     }
 
+    /** @internal */
+    public _pickMove(pointerId: number): Nullable<PickingInfo> {
+        const scene = this._scene;
+        const pickResult = scene.pick(
+            this._unTranslatedPointerX,
+            this._unTranslatedPointerY,
+            scene.pointerMovePredicate,
+            false,
+            scene.cameraToUseForPointers,
+            scene.pointerMoveTrianglePredicate
+        );
+
+        this._setCursorAndPointerOverMesh(pickResult, pointerId, scene);
+
+        return pickResult;
+    }
+
+    private _setCursorAndPointerOverMesh(pickResult: Nullable<PickingInfo>, pointerId: number, scene: Scene) {
+        const engine = scene.getEngine();
+        const canvas = engine.getInputElement();
+
+        if (pickResult && pickResult.hit && pickResult.pickedMesh) {

```

Comment:
Same is pickedResult.pickedMesh not enough ?

Reply:
Changed